tests/integration_tests/test_docker_executable.py

```diff
@@ -0,0 +1,297 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+"""Integration tests which leverage the source-faker connector to test the framework end-to-end with the Docker Executor."""
+
+from __future__ import annotations
+
+import os
+import shutil
+import tempfile
+from collections.abc import Generator
+from pathlib import Path
+
+import airbyte as ab
+import pytest
+import ulid
+from airbyte.caches.base import CacheBase
+from airbyte.caches.duckdb import DuckDBCache
+from airbyte.caches.postgres import PostgresCache
+from airbyte.caches.util import new_local_cache
+
+# Product count is always the same, regardless of faker scale.
+NUM_PRODUCTS = 100
+
+SEED_A = 1234
+SEED_B = 5678
+
+# Number of records in each of the 'users' and 'purchases' streams.
+FAKER_SCALE_A = 200
+# We want this to be different from FAKER_SCALE_A.
+FAKER_SCALE_B = 300
+
+
+# Patch PATH to include the source-faker executable.
+
+
+@pytest.fixture(scope="function")  # Each test gets a fresh source-faker instance.
+def source_faker_seed_a() -> ab.Source:
```

Comment:
We can make this name distinct and then apply skip logic centrally.

```suggestion
def source_docker_faker_seed_a() -> ab.Source:
```

Note this will also need to be renamed below in the file, whereever the fixture is used.